File: src/dataset/cpp/perfect-numbers_exercism-chrisnp/claude/ClaudeSonnet4_perfect_numbers_v2.cpp
#include "perfect_numbers.h"
#include <stdexcept>

namespace perfect_numbers {

classification classify(int n) {
    if (n <= 0) throw std::domain_error("Natural numbers only");
    
    if (n == 1) return deficient;
    
    int aliquot = 1;
    int f = 2;
    
    while (f * f <= n) {
        if (n % f == 0) {
            aliquot += f;
            int quotient = n / f;
            if (quotient != f) {
                aliquot += quotient;
                if (aliquot > n) return abundant;
            }
        }
        ++f;
    }
    
    if (aliquot < n) return deficient;
    if (aliquot > n) return abundant;
    return perfect;
}

}  // namespace perfect_numbers

File: src/dataset/typescript/perfect-numbers_exercism-wackerow/claude/ClaudeSonnet4_perfect-numbers_v2.ts
export default class PerfectNumbers {
  static classify(num: number): string {
    if (num < 1 || num % 1 !== 0) { throw 'Classification is only possible for natural numbers.'}
    
    if (num === 1) return 'deficient';
    
    let aliquotSum = 1; // 1 is always a divisor
    const sqrt = Math.sqrt(num);
    
    for (let i = 2; i <= sqrt; i++) {
      if (num % i === 0) {
        aliquotSum += i;
        if (i !== sqrt) {
          aliquotSum += num / i;
        }
      }
    }
    
    if (aliquotSum === num) return 'perfect';
    return aliquotSum < num ? 'deficient' : 'abundant';
  }
}